kernel/src/lib.rs

```diff
@@ -19,6 +19,7 @@
 #![feature(associated_type_defaults)]
 #![feature(try_with_capacity)]
 #![register_tool(component_access_control)]
+#![allow(rustdoc::private_intra_doc_links)]
```

Comment:
I was wondering if we could enable this in the command line together with the `--document-private-items` flag. It seems that `rustdoc` has a `--allow` option.
```
> rustdoc --help
[ .. ]
    -A, --allow LINT    Set lint allowed
```

Reply:
This is something that needs to be fixed, either on the Rust side or on our side. Of course, it won't be urgent to fix it. However, let's clarify this:
```rust
// Rustdoc does not render methods of public type aliases that reference private types, so links
// pointing to them cannot be resolved. See <https://github.com/rust-lang/rust/issues/132952>.
#![expect(rustdoc::private_intra_doc_links)]
```